core/state_transition.go

```diff
@@ -59,8 +59,9 @@ type StateTransition struct {
 	value      *big.Int
 	data       []byte
 	state      vm.StateDB
-
-	evm *vm.EVM
+	evm        *vm.EVM
+	vmErr      error // vmErr represents all errors occurred during the execution
```

Comment:
There's no reason to have this field, it can be returned as a function result when needed.